vispy/visuals/axis.py

```diff
@@ -392,9 +458,133 @@ def _get_tick_frac_labels(self):
             length = self.axis.pos[1] - self.axis.pos[0]  # in logical coords
             n_inches = np.sqrt(np.sum(length ** 2)) / transforms.dpi
 
-            major = _get_ticks_talbot(domain[0], domain[1], n_inches, 2)
+            major = _get_ticks_talbot(domain[0], domain[1], n_inches, density)
+
+            labels = []
+
+            if self.axis.axis_mapping is not None:
```

Comment:
There is a lot added here under this if statement. Would it be possible for you to move this into a separate function/method and then further split each `if` branch into a separate function? It looks like there is at least some duplicated code that could be simplified that way.

Reply:
Yes working on it